### PR TITLE
Tonyvanswet/avo 653 limit front end lookback period to 30 days

### PR DESCRIPTION
**Pull request by tonypls**
## Issue
AVO-653

## Description
This PR limits the frontend historical linking feature navigation to the last 30 days. There's also a [backend PR ](https://github.com/electricitymaps/electricitymaps/pull/5986) which limits the requests to a 30 day look-back and for links that are older than 30 days it links the oldest available and this PR will sync the url to the data returned. 

We should handle older links in AVO-654.

Feat. small navigation tidy

### CODE REVIEW

File: web/src/features/time/TimeController.tsx [hunk 1]
```diff
@@ -63,6 +64,25 @@ export default function TimeController({ className }: { className?: string }) {
...
 
+  // Sync the url to the datetime returned by the backend
+  useEffect(() => {
+    if (!datetimes || !urlDatetime) {
+      return;
+    }
+
+    const endDatetime = datetimes.at(-1);
+    if (!endDatetime) {
+      return;
```
Comment (cadeban) [inline web/src/features/time/TimeController.tsx:75]: nit: We might want to think about how to handle this case in the very unlikely chance we hit it

**Review by cadeban — APPROVED** [review]
lgtm

File: web/src/features/time/HistoricalTimeHeader.tsx [hunk 2]
```diff
@@ -65,10 +82,14 @@ export default function HistoricalTimeHeader() {
...
             size="sm"
             type="tertiary"
+            isDisabled={!isWithinHistoricalLimit}
             icon={
               <ChevronLeft
                 size={22}
-                className={twMerge('text-brand-green', !isHourly && 'opacity-50')}
+                className={twMerge(
+                  'text-brand-green',
+                  !isWithinHistoricalLimit && 'opacity-50'
```
Comment (VIKTORVAV99) [inline web/src/features/time/HistoricalTimeHeader.tsx:91]: Should this not become disabled on the last day/point inside the historical limit, not the first time outside it?

Comment (tonypls) [reply web/src/features/time/HistoricalTimeHeader.tsx:91]: I think it already does, I've updated the variable naming to try make this clearer

**Review by VIKTORVAV99 — APPROVED** [review]
LGTM! 🎉

**Pull request merged**